screeps-game-api/src/objects/mod.rs

```diff
@@ -217,6 +223,16 @@ pub unsafe trait RoomObjectProperties:
     }
 }
 
+/// All RoomObjects have positions.
+impl<T> HasPosition for T
+where
+    T: RoomObjectProperties,
+{
+    fn pos(&self) -> RoomPosition {
+        js_unwrap!(@{self.as_ref()}.pos)
+    }
+}
```

Comment:
I *think* this would still belong directly under the definition for `HasPosition`?

Reply:
The reasoning is that `RoomObjectProperties` isn't defined yet and its definition requires `HasPosition`.

How about we split `mod.rs` in more than one ginormous file (especially since there are still 2-3 other traits to add to the mix).

Reply:
We could definitely split it, but I think that'd be another PR. I think I'd rather keep trait implementations together with their source than have strict sequential definitions though, especially given that things can recursively depend on each other.

I've seen both putting things in order of least dependencies -> most dependencies as well as most dependencies -> least dependencies in different projects. I'm not really convinced either is as useful for code organization as keeping related things together is.